src/common/components/repository-row/dropdowns/register-name-dropdown.js

```diff
@@ -47,6 +54,60 @@ const Caption = (props) => {
   );
 
   let caption;
+  let changeForm;
+  let message;
+
+  const changeRegisteredName = !!registeredName;
+
+  // only show "change name" info and form if there is a name registered already,
+  // user is authenticated and we are not showing register success message
+  const showChangeForm = (
+    changeRegisteredName &&
+    authStore.authenticated &&
+    !registerNameStatus.success
+  );
+
+  if (!changeRegisteredName) {
+    // if user is registering new name
+    message = 'To publish to the snap store, this repo needs a registered name.';
+    if (!authStore.authenticated) {
+      message += ' You need to sign in to Ubuntu One to register a name.';
+    }
+  } else {
+    // if user is changing already registered name
+    if (!authStore.authenticated) {
+      message = 'You need to sign in to Ubuntu One to change a snap’s registered name.';
+    }
+  }
+
+  if (showChangeForm) {
+    changeForm = (
+      <div>
+        If you change the registered name:
+        <ul>
+          { snapcraftData.name === registeredName &&
+            <li>Builds will stop until you update the snapcraft.yaml to match.</li>
+          }
+          <li>The old name will remain registered to you and can be used for another snap later.</li>
+        </ul>
+        <p>
+          <label>New name:
+            {' ' /* force space between inline elements */}
+            <input
+              className={ styles.snapNameInput }
+              type='text'
+              value={ snapName }
+              onChange={ onSnapNameChange }
+            />
```

Comment:
At this point it would be hard.
Im not sure if we want to inject whole input from repository row to the dropdown (or the other way around).

With #468 coming at some point both UIs should be made consistent, so we will likely end up with input only in the dropdown, and not in table itself.